bf_simple2_yk.c

```diff
@@ -115,11 +115,22 @@ int main(int argc, char *argv[]) {
     if (cells == NULL)
         err(1, "out of memory");
 
-    YkLocation *yklocs = malloc(sizeof(YkLocation) * prog_len);
+    // FIXME: None of this is freed, but we won't get an opportunity until
+    // guards don't abort.
+    YkLocation **yklocs = calloc(prog_len, sizeof(YkLocation *));
```

Comment:
This looks weird: why are we `malloc`ing one block per `YkLocation`? Is that because of the "temporary" location transition design stuff?

Reply:
We start by allocating the NULL pointers that we pass for locations that cannot be loop starts.

Admittedly I didn't think deeply about how to optimise this. We could store a map of `instr -> location`?

Reply:
That's what the previous code did.

Reply:
The old code [allocated a `YkLocation` for each BF memory cell](https://github.com/ykjit/ykcbf/pull/8/files#diff-a374b7d1472376534b277cb7abb8b12f1759bc21ac1621f01e8cc1953bc51fc5L113). Either that's a bug, or I don't understand.

I see that `yk_simple2_yk.c` has a notion of forward and backward jumps. Perhaps you mean I could tie it into that?

Reply:
The `CELLS_LEN` thing was a bug, which you've fixed with `prog_len` -- that makes sense. But I can't see any reason to change from `YkLocation *` to `YkLocation **`.

Reply:
Fixed in b432b06 and f60deb16.